packages/opentelemetry-plugin-http/src/utils.ts

```diff
@@ -217,6 +217,46 @@ export const setSpanWithError = (
   span.setStatus(status);
 };
 
+/**
+ * Adds attributes for content-length and content-encoding HTTP headers
+ * @param { OutgoingHttpHeaders | IncomingHttpHeaders } headers http headers
+ * @param { Attributes } Attributes span attributes
+ * @param { boolean } isRequest set true for setting request content-header
+ */
+export const setContentLengthAttributes = (
+  headers: OutgoingHttpHeaders | IncomingHttpHeaders,
+  attributes: Attributes,
+  isRequest: boolean
+) => {
+  let isCompressed = false;
+
+  if (headers['content-length'] === undefined) return;
+
+  if (
+    headers['content-encoding'] &&
```

Comment:
HTTP header names are case-insensitive so you should probably make a copy of headers then iterate through all keys make them lowercase and then compare.

Reply:
It looks like the headers are already lower cased https://nodejs.org/api/http.html#http_message_headers